13/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/multinode-treepicker.md

```diff
@@ -8,45 +8,156 @@
 
 The Multinode Treepicker allows you to configure the type of tree to render and what part of the tree that should be rendered. For content it allows you to select a dynamic root node based on the current document using the multinode tree picker.
 
-**Node type:** set the type of node, the root node of the tree, or query for the root node
+### Node type
+set the type of node, the root node of the tree, or query for the root node
 
-For querying for a root node, you can use dynamic placeholders in the XPath query, following the below sample queries
+For querying content you can specify a dynamic root
+![Multinode Treepicker Data Type Definition](../../images/mntp_node_type.png)
 
-```
-// get the first textpage below the current document
-$current/textpage: current page or closest found ancestor
+When you specify the dynamic root, you have are able to navigate the tree relative to a node.
+![Multinode Treepicker Data Type Definition](../../images/mntp_node_type_dynamic_root.png)
+
+First you have to specify a origin, from where the query will start.
+![Multinode Treepicker Data Type Definition](../../images/mntp_node_type_dynamic_root_origin.png)
+
+You have the following options:
+- Root
+  - The root is the first level item of the current nodes subtree.
+- Parent
+  - The parent is the nearest ancestor of the current node.
+- Current
+    - The current node. Be aware a picker that uses the current node, cannot pick anything when the current node is created, because it do not have any children.
+- Site
+  - The nearest ancestor of the current node that has a domain assigned.
+- Specific node
+  - A specific node that you have to choose in the tree
+
+Often a origin is a good dynamic root, but you also have the option to execute multiple steps from the origin to navigate the tree to find another root.
+![Multinode Treepicker Data Type Definition](../../images/mntp_node_type_dynamic_root_steps.png)
+
+Yoy have the following options:
+- Nearest Ancestor or Self
+  -  Finds the nearest ancestor or the item itself, that fits with one of the configured document types.
+- Furthest Ancestor or Self
+  -  Finds the furthest ancestor or the item itself, that fits with one of the configured document types.
+- Nearest Descendant or Self
+  -  Finds the nearest descendant or the item itself, that fits with one of the configured document types.
+- Furthest Descendant or Self
+  -  Finds the furthest descendant or the item itself, that fits with one of the configured document types.
+- Custom
+  - Execute a custom query step by specifying the name. This requires custom code to add the new query step.
+
+Each query step takes the output from the last step (or the origin) as input.
+
+![Multinode Treepicker Data Type Definition](../../images/mntp_node_type_dynamic_root_overview.png)
+
+#### Adding a custom query step
+
+Custom query steps can be used to solve some very specific use cases.
```

Comment:
```suggestion
Custom query steps can be used to solve some specific use cases.
```

Reply:
Personally I still think "very" is important here, but at least something like "quite"

Reply:
@bergmania Fair, then lets just keep very.

One other thing, I think it would make sense to just shortly mention which specific use cases it is this custom query can solve :-)

Reply:
Only your imagination is the limitation ;) 

There is one example here https://github.com/umbraco/Umbraco-CMS/discussions/15097#discussioncomment-7520854

I think it is hard to write good examples in a short way or use example we would not recommend. But we know some people used xpath to query for content with property of a specific value.

Reply:
To be fair, that does not seem very specific to me 🙈

Maybe we could say something like: "Custom query steps can be used to solve some specific use cases." A use case could be using xpath to query for content with property of a specific value."

Reply:
I would argue, you have done something wrong if this is required. I'm just afraid of giving people bad ideas.

I'll let it be up to you, if you think is better, then feel free to do it